blaze/compute/spark.py

```diff
@@ -124,8 +128,25 @@ def compute_up(t, rdd, **kwargs):
     return rdd.distinct()
 
 
-@dispatch(Join, RDD, RDD)
-def compute_up(t, lhs, rhs, **kwargs):
+def jgetattr(data, attr, default=None):
+    """Spark's API doesn't properly implement the ``getattr`` interface, so
+    we work around it.
+    """
+    try:
+        return getattr(data, attr, default)
+    except py4j.protocol.Py4JJavaError:
+        return default
+
+
+@compute_up.register(Join, SparkDataFrame, SparkDataFrame)
+def spark_df_join(t, lhs, rhs, **kwargs):
+    # ship to rdd land, so we can reuse handling of combining records code
+    rdd = compute_up(t, lhs.rdd, rhs.rdd, **kwargs)
+    return lhs.sql_ctx.createDataFrame(rdd)
```

Comment:
we might want to eventually do this with `lhs.join(lhs.column == rhs.column, 'inner').selectExprs(*list_of_spark_Column_instances)`, I just didn't want to reimplement the `pair_assemble` logic